solr/solrj/src/java/org/apache/solr/client/solrj/impl/CloudSolrClient.java

```diff
@@ -789,7 +788,8 @@ protected NamedList<Object> requestWithRetryOnStaleState(
     if (!inputCollections.isEmpty()
         && !isAdmin
         && !isCollectionRequestOfV2) { // don't do _stateVer_ checking for admin, v2 api requests
-      Set<String> requestedCollectionNames = resolveAliases(inputCollections);
+      Set<String> requestedCollectionNames =
+          getClusterStateProvider().resolveAliases(inputCollections);
```

Comment:
I like that you are moving more alias resolution logic to the CSP, but nonetheless CloudSolrClient is still explicitly involved.  I would prefer if the matter was invisible to CloudSolrClient (make no reference here to aliases; let it be a detail in CSP).  Admittedly that might be too much for this issue/PR.